src/main/java/study/racingcar/domain/RacingGame.java

```diff
@@ -0,0 +1,34 @@
+package study.racingcar.domain;
+
+import java.util.List;
+
+public class RacingGame {
+    private Cars cars;
+    private MovableDistance movableDistance;
+
+    public RacingGame(Cars cars,
+                      MovableDistance movableDistance) {
+        this.cars = cars;
+        this.movableDistance = movableDistance;
+    }
+
+    public RacingGame(List<String> carNames, MovableDistance movableDistance) {
+        this(new Cars(carNames), movableDistance);
+    }
+
+    public RacingGameResult play(int time) {
+        return this.play(new Time(time));
+    }
+
+    public RacingGameResult play(Time time) {
+        GameEvent gameEvent = new GameEvent();
+
+        for (int i = 0, count = time.getTime(); i < count; i++) {
+            gameEvent.add(cars.clone());
+            cars = gameEvent.getLastEvent();
+            cars.move(movableDistance);
+        }
+
+        return new RacingGameResult(gameEvent);
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        List<Cars> results = new ArrayList<>(cars.clone());

        for (int i = 0, count = time.getTime(); i < count; i++) {
            cars.move(movableDistance);
            results.add(cars.clone());
        }

        return new RacingGameResult(results);
```